app/views/adventures/offline.html.erb

```diff
@@ -0,0 +1,25 @@
+<% content_for :theme, "#{@adventure.theme.capitalize}Theme" %>
+
+<div class="PlayerActions">
+<% if @adventure.editable_by?(current_user) %>
+  <%= link_to 'Edit', [:edit, @adventure], class: "SlantButton" %>
+<% else %>
+  <%= link_to 'Source', [:source, @adventure], class: "SlantButton" %>
+<% end %>
+
+<%= link_to 'Save', [:offline, @adventure], class: "SlantButton", download: @adventure.title%>
```

Comment:
My hunch is that this will kick off some javascript which replicates the cmd-s event instead of hitting the server. If such a thing is possible.

Reply:
Yeah so I have found a few options, I was checking out the ruby `send_data` function, which would work with sending just the html. I considered finding a way to call that multiple times and maybe creating a zip for all of the assets and sending that. The cmd-s event downloads like this: 
![image](https://user-images.githubusercontent.com/16651533/76227606-f0059180-61f5-11ea-9fd7-dc4a9614459c.png)

Not sure if this is not the way forward though. I found another possibility is using the new HTML5 `download` attribute for <a> tags. Problem with that being that I don't think there is any way to include assets in that, so again it downloads only the `.htm` file, which looks like this: 
![image](https://user-images.githubusercontent.com/16651533/76227936-6a361600-61f6-11ea-98ca-9dab12ce994c.png)

The final way forward could be using JS that executes on hitting the button -- I have a lot of tries at all three of these ways and did not get really lucky with it. I think the JS way would likely need to be similar to the ruby `send_data`, where I would need to get the assets too, maybe zip it up, and then trigger a download. I do not think it's possible for the button to execute cmd-s, so we'd need to just mimic it as closely as possible